### PR TITLE
Bomb item

### LINKED ISSUE #58: Bomb item

### PR DESCRIPTION
**Pull request by AideTechBot**
Fixes #58 and block dupe glitch

### CODE REVIEW

File: Assets/Scripts/RedRunner/Spawner/BombExplosion.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+using System.Collections.Generic;
+using UnityEngine;
+
+public class BombExplosion : MonoBehaviour
+{
+    [SerializeField]
+    protected float m_ExplodeRadius = 10.0f;
+
+    // Start is called before the first frame update
+    void Start()
```
Comment (srhickma) [inline Assets/Scripts/RedRunner/Spawner/BombExplosion.cs]: Remove this

File: Assets/Scripts/RedRunner/Spawner/BombExplosion.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    {
+        
+    }
+
+    // boom goes the dynamite
+    public void Explode()
+    {
+        Collider2D[] hitColliders = Physics2D.OverlapCircleAll(transform.position, m_ExplodeRadius);
+        for (int i = 0; i < hitColliders.Length; i++)
```
Comment (srhickma) [inline Assets/Scripts/RedRunner/Spawner/BombExplosion.cs]: Nit: `foreach (var collider in hitColliders) ...`

File: Assets/Scripts/RedRunner/Spawner/BombExplosion.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+                    curr = curr.parent;
+                }
+                Destroy(curr.gameObject);
+            }
+        }
+
+    }
+
+    // Update is called once per frame
+    void Update()
```
Comment (srhickma) [inline Assets/Scripts/RedRunner/Spawner/BombExplosion.cs]: Remove this

File: Assets/Scripts/RedRunner/Spawner/Deletable.cs [hunk 4]
```diff
@@ -0,0 +1,18 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Deletable : MonoBehaviour
+{
+    // Start is called before the first frame update
+    void Start()
```
Comment (srhickma) [inline Assets/Scripts/RedRunner/Spawner/Deletable.cs]: Remove this

File: Assets/Scripts/RedRunner/Spawner/Deletable.cs [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    {
+        
+    }
+
+    // Update is called once per frame
+    void Update()
```
Comment (srhickma) [inline Assets/Scripts/RedRunner/Spawner/Deletable.cs]: Remove this

**Review by srhickma — APPROVED** [review]
LGTM with comments

File: Assets/Prefabs/Spawn Blocks/SpawnBlock_1.prefab [hunk 6]
```diff
@@ -93,17 +93,25 @@ MonoBehaviour:
   m_EditorClassIdentifier: 
   m_Width: 0
   m_Probability: 1
+--- !u!114 &594285076651998136
```
Comment (WilsonCWu) [inline Assets/Prefabs/Spawn Blocks/SpawnBlock_1.prefab:96]: Double checking, what is the reason for all these changes?

File: Assets/Scripts/RedRunner/Spawner/BombExplosion.cs [hunk 7]
```diff
@@ -0,0 +1,43 @@
+﻿using RedRunner.Networking;
+using RedRunner.TerrainGeneration;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class BombExplosion : MonoBehaviour
+{
+    [SerializeField]
+    protected float m_ExplodeRadius = 10.0f;
```
Comment (WilsonCWu) [inline Assets/Scripts/RedRunner/Spawner/BombExplosion.cs:10]: does the UI circle generate based off this value too?

**Review by WilsonCWu — COMMENTED** [review]
Is it worthwhile to do the animation in this PR too, so an explosion animation plays on delete?

**Pull request merged**